chains/ethereum/state_checks.go

```diff
@@ -0,0 +1,28 @@
+package ethereum
+
+import (
+	"math/big"
+
+	"github.com/ChainSafe/log15"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+)
+
+// Queries the contract for the current deposit status
+func (w *Writer) DepositStatus(depositId *big.Int, originChain *big.Int) (uint8, error) {
+	ret := new(struct {
```

Comment:
can we convert this struct to a type for consistency across the bridge

Reply:
I've created a DepositProposal Type within the `state_check.go` file